core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/AddExchangesBelowPartialAggregationOverGroupIdRuleSet.java

```diff
@@ -247,6 +262,25 @@ protected Optional<PlanNode> transform(AggregationNode aggregation, GroupIdNode
                     .map(groupId.getGroupingColumns()::get)
                     .collect(toImmutableList());
 
+            if (isUseHighestCardinalityColumnForForcedExchangeBelowGroupId(context.getSession())) {
+                // use only the symbol with the highest cardinality.
+                // this makes partial aggregation more efficient in case of low correlation between symbols
```

Comment:
nit: Probably it would make sense to look at actual NDV of `maxNdvSymbol`. In case of low NDV, it might make sense to add more symbols to group by (especially if these are not correlated)